src/path/fs.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2018 The predicates-rs Project Developers.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::fmt;
+use std::path;
+
+use path::{FileContent, FileTypePredicate};
+use Predicate;
+
+/// Predicate that compares file matches
+#[derive(Clone, Debug)]
+pub struct FileSystemEntryPredicate {
+    file_type: FileTypePredicate,
+    file_content: Option<FileContent>,
+}
+
+impl Predicate<path::Path> for FileSystemEntryPredicate {
+    fn eval(&self, path: &path::Path) -> bool {
+        if self.file_type.is_file() {
+            if !self.file_type.eval(path) {
+                false
+            } else {
+                self.file_content.clone().unwrap() == FileContent::new(path).unwrap()
```

Comment:
`self.file_content.as_ref().unwrap()` addresses the `clone`

The first `unwrap` should probably be an `expect` so you can document your assumption of why it is always safe to `unwrap` in this context.

Should the second `unwrap`  instead have us return `false`?